src/dos/cdrom.cpp

```diff
@@ -18,6 +18,217 @@
 
 #include "cdrom.h"
 
+#include "../libs/sdlcd/SDL_cdrom.c"
+#if defined(WIN32)
+#define SDL_CDROM_WIN32
+#include "../libs/sdlcd/win32/SDL_syscdrom.c"
+#elif defined(LINUX)
+#define SDL_CDROM_LINUX
+#include "../libs/sdlcd/linux/SDL_syscdrom.c"
+#elif defined(MACOSX) && defined(__clang__)
+#define SDL_CDROM_MACOSX
+#include "../libs/sdlcd/macosx/SDL_syscdrom.c"
+#include "../libs/sdlcd/macosx/AudioFilePlayer.c"
+#include "../libs/sdlcd/macosx/AudioFileReaderThread.c"
+#include "../libs/sdlcd/macosx/CDPlayer.c"
+#include "../libs/sdlcd/macosx/SDLOSXCAGuard.c"
+#else
+#define SDL_CDROM_DUMMY
+#include "../libs/sdlcd/dummy/SDL_syscdrom.c"
+#endif
+
+// ******************************************************
+// SDL CDROM
+// ******************************************************
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "dosbox.h"
+#include "SDL.h"
+#include "support.h"
+#include "cdrom.h"
+
+CDROM_Interface_SDL::CDROM_Interface_SDL(void)
+{
+	driveID = 0;
+	oldLeadOut = 0;
+	cd = 0;
+}
+
+CDROM_Interface_SDL::~CDROM_Interface_SDL(void)
+{
+	StopAudio();
+	SDL_CDClose(cd);
+	cd = 0;
+}
+
+bool CDROM_Interface_SDL::SetDevice(char *path, int forceCD)
+{
+	char buffer[512];
+	strcpy(buffer, path);
+	upcase(buffer);
```

Comment:
Looking at the code, `buffer` could probably be replaced with `std::string`. There's an `upcase` overload that supports `std::string`.

Reply:
Done.